src/components/list/moto_list.tsx

```diff
@@ -106,18 +163,16 @@ export default function MotorcycleList({ motorcycles, setMotorcycles }: Motorcyc
                 value={editMotorcycle?.engine || ''}
                 onChange={(e) => setEditMotorcycle({ ...editMotorcycle!, engine: e.target.value })}
               >
-                {engineTypes.map((engine) => (
-                  <FormControlLabel key={engine} value={engine} control={<Radio />} label={engine} />
-                ))}
+                {engineTypes.map((engine) => (<FormControlLabel checked={engine == watch('engine')} {...register('engine', {required: 'Engine type is required'})} value={engine} control={<Radio />} label={engine} />))}
               </RadioGroup>
             </FormControl>
           </Stack>
         </DialogContent>
         <DialogActions>
           <Button onClick={handleEditClose}>Cancel</Button>
-          <Button onClick={handleSaveEdit}>Save</Button>
+          <Button onClick={handleSaveEdit} disabled={hasErrors}>Save</Button>
```

Comment:
After you use `handleSubmit` from react-hook-form (see prev comment), you don't need to set `onClick` to this button, but just make it `type="submit"`. This way it will automatically trigger the submit function that you set to your container, and executing the validation